src/gmpints.c

```diff
@@ -481,21 +567,18 @@ void PrintInt ( Obj op )
     CHECK_INT(op);
 
     /* use gmp func to print int to buffer                                 */
-    if (!IS_INTPOS(op)) {
-      buf[0] ='-';
-      signlength = 1;
-    } else {
-      signlength = 0;
-    }
-    gmp_snprintf((char *)(buf+signlength),20000-signlength,
-                 "%Nd", ADDR_INT(op),
-                 (TypGMPSize)SIZE_INT(op));
+    Char buf[20000];
+    mpz_t v;
```

Comment:
Note that the `assert` was not even doing anything by default.

Anyway, I'll add a comment and think about whether I can make this more clear / pseudo-robust.